### PR TITLE
Fix release actions

### PR DESCRIPTION
**Pull request by anishasthana**
This is related to #1133 . The PR does the following:

1) Update the release job names to (accurately) refer to apiserver instead of operator.
2) Update the documentation to have references to the quay.io images.

### CODE REVIEW

**Comment by anishasthana** [comment]
cc @kevin85421

File: docs/deploy/images.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+Images for the various KubeRay components are published at the following locations:
+
+1. [Quay.io](https://quay.io/organization/kuberay)
+2. [Dockerhub](https://hub.docker.com/u/kuberay)
+
+We recommend using Quay as the primary source for images as there are are no image-pull restrictions on Quay.io.
+
+## Stable versions
+
+For stable releases, use version tags (e.g. `kuberay/operator:v0.6.0`).
```
Comment (kevin85421) [inline docs/deploy/images.md:12]: Will `kuberay/operator:v0.6.0` pull images from Quay by default or should we change to `quay.io/kuberay/operator:v0.6.0`?

File: docs/deploy/images.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+## Master commits
+
+The first seven characters of the git SHA specify images built from specific commits
+(e.g. `kuberay/operator:944a042`).
```
Comment (kevin85421) [inline docs/deploy/images.md:17]: same

File: docs/deploy/images.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+## Nightly images
+
+The nightly tag specifies images built from the most recent master (e.g. `kuberay/operator:nightly`).
```
Comment (kevin85421) [inline docs/deploy/images.md:21]: same

File: docs/deploy/images.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# Container Images
+
+Images for the various KubeRay components are published at the following locations:
+
+1. [Quay.io](https://quay.io/organization/kuberay)
+2. [Dockerhub](https://hub.docker.com/u/kuberay)
+
+We recommend using Quay as the primary source for images as there are are no image-pull restrictions on Quay.io.
```
Comment (kevin85421) [inline docs/deploy/images.md:8]: Would you mind adding more details about "image-pull restrictions"?

File: .github/workflows/image-release.yaml [hunk 5]
```diff
@@ -76,7 +76,7 @@ jobs:
         username: ${{ secrets.QUAY_USERNAME }}
         password: ${{ secrets.QUAY_ROBOT_TOKEN }}
 
-    - name: Push Operator to Quay.io
+    - name: Push Apiserver to Quay.io
```
Comment (kevin85421) [inline .github/workflows/image-release.yaml:79]: Good catch!

File: docs/deploy/images.md (hunk 1 above)
Comment (anishasthana) [reply docs/deploy/images.md:12]: We will need to update that -- the reason I didn't mention it was that the image isn't available on Quay yet. If you can push the images up there, I can update the docs accordingly!

Comment (kevin85421) [reply docs/deploy/images.md:12]: Done. I uploaded both operator and apiserver images to Quay manually.

Comment (anishasthana) [reply docs/deploy/images.md]: Updated

File: docs/deploy/images.md (hunk 2 above)
Comment (anishasthana) [reply docs/deploy/images.md]: Updated

File: docs/deploy/images.md (hunk 4 above)
Comment (anishasthana) [reply docs/deploy/images.md]: Updated

**Pull request merged**